cilium-dbg/cmd/kvstore.go

```diff
@@ -47,9 +47,17 @@ func setupKvstore(ctx context.Context, logger *slog.Logger) {
 		}
 	}
 
-	if err := kvstore.Setup(ctx, logger, kvStore, kvStoreOpts, nil); err != nil {
-		Fatalf("Unable to setup kvstore: %s", err)
+	client, errch := kvstore.NewClient(ctx, logger, kvStore, kvStoreOpts, nil)
+	select {
+	case <-ctx.Done():
```

Comment:
kvstore.NewClient doesn't respect the passed context ?

Reply:
It does respect the context, but I'm not 100% sure it actually closes the error channel in all cases the context is canceled. I'd prefer to be a bit more explicit here for the moment, pending subsequent refactoring to rework that logic to be more straightforward to reason about.